core/shard_thread.go

```diff
@@ -0,0 +1,136 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/dicedb/dice/config"
+	"github.com/dicedb/dice/core/diceerrors"
+	"github.com/dicedb/dice/server/utils"
+)
+
+type ShardID int8
+
+type StoreOp struct {
```

Comment:
StoreOperation related data-structures are both common to Worker and Shard. IMO it is better to have them in a separate package like `ops` which would avoid cyclic dependency in future.

Reply:
Done, this required us to move a lot of other structs into their own packages (client, session, RedisCmd). I think this is a good move towards more modularized code.